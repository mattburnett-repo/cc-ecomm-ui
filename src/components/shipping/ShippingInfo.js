export default function ShippingInfo () {
    return (
        <div>
            <div>
                <label htmlFor="first-name">First Name:</label>
                <input id='first-name' aria-label="first-name"/>
            </div>
            <div>
                <label htmlFor="last-name">Last Name:</label>
                <input id='last-name' aria-label="last-name"/>
            </div>
            <div>
                <label htmlFor="address-1">Address 1:</label>
                <input id='address-1' aria-label="address-1"/>
            </div>
            <div>
                <label htmlFor="address-2">Address 2:</label>
                <input id='address-2' aria-label="address-2"/>
            </div>
            <div>
                <label htmlFor="city">City:</label>
                <input id='city' aria-label="city"/>
            </div>
            <div>
                <label htmlFor="state-province">State / Province:</label>
                <input id='state-province' aria-label="state-province"/>
            </div>
            <div>
                <label htmlFor="postal-code">Postal Code:</label>
                <input id='postal-code' aria-label="postal-code"/>
            </div>
            <div>
                <label htmlFor="country">Country:</label>
                <input id='country' aria-label="country"/>
            </div>
            <button aria-label="go-to-payment-info">Go To Payment Info</button>
        </div>
    )
}