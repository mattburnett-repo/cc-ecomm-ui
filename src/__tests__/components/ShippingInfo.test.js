// 20211217: Wireframe https://wireframe.cc/ag82hc

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import ShippingInfo from '../../components/shipping/ShippingInfo';

describe('<ShippingInfo /> component tests', () => {
    beforeEach(() => render(<ShippingInfo />));

    it('should display first name', () => {
        screen.getByRole('textbox', {name: /first-name/i});
    });
    it('should display last name', () => {
        screen.getByRole('textbox', {name: /last-name/i});
    });
    it('should display address 1', () => {
        screen.getByRole('textbox', {name: /address-1/i});
    });
    it('should display address 2', () => {
        screen.getByRole('textbox', {name: /address-2/i});
    });
    it('should display city', () => {
        screen.getByRole('textbox', {name: /city/i});
    });
    it('should display state / province', () => {
        screen.getByRole('textbox', {name: /state-province/i});
    });
    it('should display postal code', () => {
        screen.getByRole('textbox', {name: /postal-code/i});
    });
    it('should display country', () => {
        screen.getByRole('textbox', {name: /country/i});
    });
    it('should display go-to-payment-info button', () => {
        screen.getByRole('button', {name: /go-to-payment-info/i});
    });
});

describe('<ShippingInfo /> feature tests', () => {
    beforeEach(() => render(<ShippingInfo />));

    it('accepts first name', () => {
        const theElement = screen.getByRole('textbox', {name: /first-name/i});
        userEvent.type(theElement, 'first-name');
        
        expect(theElement).toHaveValue('first-name');
    });
    it('accepts last name', () => {
        const theElement = screen.getByRole('textbox', {name: /last-name/i});
        userEvent.type(theElement, 'last-name');
        
        expect(theElement).toHaveValue('last-name');
    });
    it('accepts address 1', () => {
        const theElement = screen.getByRole('textbox', {name: /address-1/i});
        userEvent.type(theElement, 'address-1');
        
        expect(theElement).toHaveValue('address-1');
    });
    it('accepts address 2', () => {
        const theElement = screen.getByRole('textbox', {name: /address-2/i});
        userEvent.type(theElement, 'address-2');
        
        expect(theElement).toHaveValue('address-2');
    });
    it('accepts city', () => {
        const theElement = screen.getByRole('textbox', {name: /city/i});
        userEvent.type(theElement, 'city');
        
        expect(theElement).toHaveValue('city');
    });
    it('accepts state / province', () => {
        const theElement = screen.getByRole('textbox', {name: /state-province/i});
        userEvent.type(theElement, 'state-province');
        
        expect(theElement).toHaveValue('state-province');
    });
    it('accepts postal code', () => {
        const theElement = screen.getByRole('textbox', {name: /postal-code/i});
        userEvent.type(theElement, 'postal-code');
        
        expect(theElement).toHaveValue('postal-code');
    });
    it('accepts country', () => {
        const theElement = screen.getByRole('textbox', {name: /country/i});
        userEvent.type(theElement, 'country');
        
        expect(theElement).toHaveValue('country');
    });

    it('creates a snapshot', () => {
        const tree = renderer.create(<ShippingInfo />).toJSON();
        expect(tree).toMatchSnapshot();  
    });
});

describe('<ShippingInfo /> unit / functional tests', () => {
    beforeEach(() => render(<ShippingInfo />));

    it('clicks the go-to-payment-info button', () => {
        const theVal = screen.getByRole('button', {name: /go-to-payment-info/i});
        fireEvent.click(theVal);
    });
})
