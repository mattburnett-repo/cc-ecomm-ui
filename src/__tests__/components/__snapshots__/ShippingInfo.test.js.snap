// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingInfo /> component tests creates a snapshot 1`] = `
<div
  aria-label="shipping-info"
  role="presentation"
>
  <div>
    <label
      htmlFor="first-name"
    >
      First Name:
    </label>
    <input
      aria-label="first-name"
      id="first-name"
    />
  </div>
  <div>
    <label
      htmlFor="last-name"
    >
      Last Name:
    </label>
    <input
      aria-label="last-name"
      id="last-name"
    />
  </div>
  <div>
    <label
      htmlFor="address-1"
    >
      Address 1:
    </label>
    <input
      aria-label="address-1"
      id="address-1"
    />
  </div>
  <div>
    <label
      htmlFor="address-2"
    >
      Address 2:
    </label>
    <input
      aria-label="address-2"
      id="address-2"
    />
  </div>
  <div>
    <label
      htmlFor="city"
    >
      City:
    </label>
    <input
      aria-label="city"
      id="city"
    />
  </div>
  <div>
    <label
      htmlFor="state-province"
    >
      State / Province:
    </label>
    <input
      aria-label="state-province"
      id="state-province"
    />
  </div>
  <div>
    <label
      htmlFor="postal-code"
    >
      Postal Code:
    </label>
    <input
      aria-label="postal-code"
      id="postal-code"
    />
  </div>
  <div>
    <label
      htmlFor="country"
    >
      Country:
    </label>
    <input
      aria-label="country"
      id="country"
    />
  </div>
  <button
    aria-label="go-to-payment-info"
  >
    Go To Payment Info
  </button>
</div>
`;

exports[`<ShippingInfo /> feature tests creates a snapshot 1`] = `
<div>
  <div>
    <label
      htmlFor="first-name"
    >
      First Name:
    </label>
    <input
      aria-label="first-name"
      id="first-name"
    />
  </div>
  <div>
    <label
      htmlFor="last-name"
    >
      Last Name:
    </label>
    <input
      aria-label="last-name"
      id="last-name"
    />
  </div>
  <div>
    <label
      htmlFor="address-1"
    >
      Address 1:
    </label>
    <input
      aria-label="address-1"
      id="address-1"
    />
  </div>
  <div>
    <label
      htmlFor="address-2"
    >
      Address 2:
    </label>
    <input
      aria-label="address-2"
      id="address-2"
    />
  </div>
  <div>
    <label
      htmlFor="city"
    >
      City:
    </label>
    <input
      aria-label="city"
      id="city"
    />
  </div>
  <div>
    <label
      htmlFor="state-province"
    >
      State / Province:
    </label>
    <input
      aria-label="state-province"
      id="state-province"
    />
  </div>
  <div>
    <label
      htmlFor="postal-code"
    >
      Postal Code:
    </label>
    <input
      aria-label="postal-code"
      id="postal-code"
    />
  </div>
  <div>
    <label
      htmlFor="country"
    >
      Country:
    </label>
    <input
      aria-label="country"
      id="country"
    />
  </div>
  <button
    aria-label="go-to-payment-info"
  >
    Go To Payment Info
  </button>
</div>
`;
